torch/_dynamo/variables/functions.py

```diff
@@ -444,6 +444,12 @@ def _setup_exception(self, tx, exc):
             # exception is raised again.
             tracer.exception_handler(e)
 
+    def _is_inside_try_finally(self, tracer):
+        exn_tab_entry = tracer.current_instruction.exn_tab_entry
+        return (sys.version_info >= (3, 11) and exn_tab_entry) or (
+            sys.version_info < (3, 11) and tracer.block_stack
```

Comment:
@williamwen42 it looks like you implemented the exn_tab_entry. Could you confirm this function is correct?

Reply:
A presence of `exn_tab_entry` means we need to jump somewhere if an exception is encountered. Instructions in `try` and `with` blocks have an `exn_tab_entry`, but other instructions can also have `exn_tab_entry` in other cases (e.g. `with` block cleanup code).

try-except with no finally will still have `exn_tab_entry`s for the instructions in the try block.

If the intended behavior of this function is "return True if we're in a region where we should be jumping when an exception is encountered", then this function looks correct.

Reply:
I think this one is wrong. It should return if there is a finally block

Reply:
In general, especially for 3.11+, it's difficult to tell just from looking at bytecode where the try/except/finally boundaries are.